src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -785,21 +788,34 @@ private void EnsureODataFormatAndContentType()
 
                 if (this.writerPayloadKind == ODataPayloadKind.Batch)
                 {
-                    // Note that this serves as verification only for now, since we only support a single content type and format for $batch payloads.
-                    Debug.Assert(this.format == ODataFormat.Batch, "$batch should only support batch format since it's format independent.");
-                    Debug.Assert(this.mediaType.FullTypeName == MimeConstants.MimeMultipartMixed, "$batch content type is currently only supported to be multipart/mixed.");
+                    Debug.Assert(this.format == ODataFormat.Batch || this.format == ODataFormat.Json,
+                        "$batch payload should only support batch or json format.");
+                    Debug.Assert(
+                        (this.format == ODataFormat.Batch && this.mediaType.FullTypeName == MimeConstants.MimeMultipartMixed)
+                     || (this.format == ODataFormat.Json && this.mediaType.FullTypeName == MimeConstants.MimeApplicationJson),
+                     "$batch content type is currently can only be multipart/mixed or application/json.");

```

Comment:
It is debug check for design invariant only. ODataMediaTypeResolver already enforced that these are the only two format/type combination that are supported. Json format is added here: https://github.com/biaol-odata/odata.net/blob/maintenance-6.x_JsonFormatBatch-A/src/Microsoft.OData.Core/ODataMediaTypeResolver.cs#L364